pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -196,6 +215,13 @@ protected void handleLookup(CommandLookupTopic lookup) {
         if (lookupSemaphore.tryAcquire()) {
             final String originalPrincipal = lookup.hasOriginalPrincipal() ? lookup.getOriginalPrincipal()
                     : this.originalPrincipal;
+            if (!validateOriginalPrincipal(originalPrincipal,
+                    newLookupErrorResponse(ServerError.AuthorizationError,
```

Comment:
We cannot create the buffer in all cases, otherwise this is being leaked when there is no error.